typing/typemod.ml

```diff
@@ -1017,27 +1060,18 @@ let rec apply_modalities_signature ~recursive env modalities sg =
       in
       let vd = {vd with val_modalities} in
       Sig_value (id, vd, vis)
-  | Sig_module (id, pres, md, rec_, vis) when recursive ->
-      let md_type = apply_modalities_module_type env modalities md.md_type in
-      let md = {md with md_type} in
+  | Sig_module (id, pres, md, rec_, vis) ->
+      let md_modalities =
+        md.md_modalities
+        |> Mode.Modality.Value.to_const_exn
+        |> (fun then_ -> Mode.Modality.Value.Const.concat ~then_ modalities)
+        |> Mode.Modality.Value.of_const
+      in
+      let md = {md with md_modalities} in
```

Comment:
As mentioned in #3943 , now `include S @@ modalities` is properly implemented, we can mention it in `syntax.md`.